### ISSUE TITLE
Sort with per-key order

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#4945]
## tl;dr

The `sort` operator in Zed currently lacks the ability to specify order on a per-key basis.

## Details

At the time this issue is being filed, Zed it at commit b7d3b41.

This limitation was recently highlighted in a conversation with community user @zmajeed.

Consider this UNIX/Linux sort that's inspired by the examples from [this Stack Overflow post](https://stackoverflow.com/questions/357560/sorting-multiple-keys-with-unix-sort):

```
$ cat sort.csv
field1,field2,field3,field4,field5
2,3,a,9,C
2,2,b,20,Fa
2,2,c,19,Gb
2,2,c,19,Ga
2,2,b,22,Ga
1,10,b,22,Ga

$ tail -n +2 sort.csv | sort -t, -k1,1n -k2,2nr
1,10,b,22,Ga
2,3,a,9,C
2,2,b,20,Fa
2,2,b,22,Ga
2,2,c,19,Ga
2,2,c,19,Gb
```

Notice how the first column is sorted by acending numeric order and then the second column is sorted in descending numeric order when there's "ties" in the first column.

In Zed there's not currently a way to achieve this in a single call to the `sort` operator: The `-r` option in Zed is applied to the totality of keys and can't be simultaneously applied to treatment of different keys like is being done above with the `-k2,2nr` to get the reverse order in the second key.

```
$ zq -version
Version: v1.11.1-28-gb7d3b41c

$ zq -f csv 'sort field1,field2' sort.csv
field1,field2,field3,field4,field5
1,10,b,22,Ga
2,2,b,20,Fa
2,2,c,19,Gb
2,2,c,19,Ga
2,2,b,22,Ga
2,3,a,9,C

$ zq -f csv 'sort -r field1,field2' sort.csv
field1,field2,field3,field4,field5
2,3,a,9,C
2,2,b,20,Fa
2,2,c,19,Gb
2,2,c,19,Ga
2,2,b,22,Ga
1,10,b,22,Ga
```

**Comment by philrz** [comment]
Another user recently asked about this in a recent [community Slack thread](https://brimdata.slack.com/archives/CTSMAK6G7/p1722310037204649). In their own words:

> can sort directions be per field?

**Comment by philrz** [comment]
Verified in Zed commit 1ffe14f.

Revisiting the example above where `zq` could not previously sort by the keys in different orders, here's how that can be achieved with the new syntax:

```
$ zq -version
Version: v1.17.0-48-g1ffe14f6

$ zq -f csv 'sort field1 asc, field2 desc' sort.csv 
field1,field2,field3,field4,field5
1,10,b,22,Ga
2,3,a,9,C
2,2,b,20,Fa
2,2,c,19,Gb
2,2,c,19,Ga
2,2,b,22,Ga
```

Closely examining the output, we can see that it's not 100% the same as in the Linux `sort` example shown for comparison. @nwt educated me that this is because Zed's `sort` is "stable", meaning that values with identical sort keys always have the same relative order in the output as they had in the input. The Linux `sort` utility does not behave this way by default but `-s` makes it stable, and indeed by adding this we can now see the same output that we just showed from `zq`.

```
$ tail -n +2 sort.csv | sort -s -t, -k1,1n -k2,2nr
1,10,b,22,Ga
2,3,a,9,C
2,2,b,20,Fa
2,2,c,19,Gb
2,2,c,19,Ga
2,2,b,22,Ga
```

The Zed [`sort` docs](https://zed.brimdata.io/docs/next/language/operators/sort) are actually in need of an update to reflect the new per-key syntax, so I'll make those revisions shortly and plan to divulge the "stable" behavior as well just in case users notice it and need an explanation.

Thanks @mattnibs and @nwt!
